### PR TITLE
initial implementation of PH2012, tests must have timeout

### PR DESCRIPTION
**Pull request by mayerj**

### CODE REVIEW

File: Philips.CodeAnalysis.MsTestAnalyzers/TestHasTimeoutCodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+					}
+					else if (attributesyntax.Name.ToString().Contains(@"TestCategory"))
+					{
+						category = attributesyntax.ArgumentList.Arguments.First().ToString();
+					}
+				}
+			}
+
+			switch (category)
+			{
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MsTestAnalyzers/TestHasTimeoutCodeFixProvider.cs:72]: These are arguably weird values for open source.

**Review by bcollamore — APPROVED** [review]

**Comment by bcollamore** [comment]
Rev and documentation

File: Philips.CodeAnalysis.MsTestAnalyzers/TestHasTimeoutCodeFixProvider.cs (hunk 1 above)
Comment (mayerj) [reply Philips.CodeAnalysis.MsTestAnalyzers/TestHasTimeoutCodeFixProvider.cs:72]: Agreed.  Let me think about it.

**Comment by mayerj** [comment]
> Rev and documentation

It was already documented as existing 👍

File: Philips.CodeAnalysis.MsTestAnalyzers/TestHasTimeoutCodeFixProvider.cs (hunk 1 above)
Comment (mayerj) [reply Philips.CodeAnalysis.MsTestAnalyzers/TestHasTimeoutCodeFixProvider.cs]: Changed behavior.  Categories and acceptable test timeouts are listed in the editorconfig.  If you have the analyzer enabled, but have a category not in the editorconfig it sets the timeout to 1000.

**Review by bcollamore — APPROVED** [review]

**Pull request merged**